packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
+                @NonNull CameraCaptureSession session,
+                @NonNull CaptureRequest request,
+                @NonNull CaptureResult partialResult) {
+              Log.i(TAG, "onCaptureProgressed");
+            }
+
             @Override
             public void onCaptureCompleted(
                 @NonNull CameraCaptureSession session,
                 @NonNull CaptureRequest request,
                 @NonNull TotalCaptureResult result) {
+              Log.i(TAG, "onCaptureCompleted");
               unlockAutoFocus();
             }
-          },
-          null);
+          };
+
+      captureSession.stopRepeating();
+      captureSession.abortCaptures();
+      Log.i(TAG, "sending capture request");
+      captureSession.capture(stillBuilder.build(), captureCallback, backgroundHandler);
     } catch (CameraAccessException e) {
-      pictureCaptureRequest.error("cameraAccess", e.getMessage(), null);
+      dartMessenger.error(flutterResult, "cameraAccess", e.getMessage(), null);
     }
   }
 
-  private void lockAutoFocus(CaptureCallback callback) {
-    captureRequestBuilder.set(
+  @SuppressWarnings("deprecation")
+  private Display getDefaultDisplay() {
+    return activity.getWindowManager().getDefaultDisplay();
+  }
+
+  /** Starts a background thread and its {@link Handler}. TODO: call when activity resumed */
+  private void startBackgroundThread() {
+    backgroundHandlerThread = new HandlerThread("CameraBackground");
+    backgroundHandlerThread.start();
+    backgroundHandler = new Handler(backgroundHandlerThread.getLooper());
+  }
+
+  /** Stops the background thread and its {@link Handler}. TODO: call when activity paused */
+  private void stopBackgroundThread() {
+    try {
+      if (backgroundHandlerThread != null) {
+        backgroundHandlerThread.quitSafely();
+        backgroundHandlerThread.join();
+        backgroundHandlerThread = null;
+      }
+
+      backgroundHandler = null;
+    } catch (InterruptedException e) {
+      dartMessenger.error(flutterResult, "cameraAccess", e.getMessage(), null);
+    }
+  }
+
+  /** Start capturing a picture, doing autofocus first. */
+  private void runPictureAutoFocus() {
+    Log.i(TAG, "runPictureAutoFocus");
+
+    cameraCaptureCallback.setCameraState(CameraState.STATE_WAITING_FOCUS);
+    lockAutoFocus();
+  }
+
+  /**
+   * Start the autofocus routine on the current capture request and calls the onCompleted callback.
+   */
+  private void lockAutoFocus() {
+    Log.i(TAG, "lockAutoFocus");
+
+    // Trigger AF to start
+    previewRequestBuilder.set(
         CaptureRequest.CONTROL_AF_TRIGGER, CaptureRequest.CONTROL_AF_TRIGGER_START);
 
-    refreshPreviewCaptureSession(
-        null, (code, message) -> pictureCaptureRequest.error(code, message, null));
+    try {
+      captureSession.capture(previewRequestBuilder.build(), null, backgroundHandler);
+    } catch (CameraAccessException e) {
+      Log.i(TAG, "Error unlocking focus: " + e.getMessage());
```

Comment:
is this an info or an error?

Reply:
This would be an error but dart would handle the error, so the log here can probably be removed, or we should feed the details back to dart.